### ISSUE TITLE
kOS 1.1.4: kOS.Safe.Exceptions.KOSCompileException: ProgramBuilder.ReplaceLabels

### ISSUE THREAD

**Issue by pellinor0** [KSP-KOS/KOS#2203]
I got this error when upgrading to kOS 1.1.4. 

I have boiled it down to 3 functions that are defined in a file tmp.ks. When I try to run that file, kOS gets caught in a loop of errors, and KSP becomes unresponsive. The only way to get out is by closing the window running KSP. 

When I remove one of those functions, the script runs without error. 

( Links removed, since they seem to be expired )

Reproduction: in KSP 1.3.1 with only kOS and ModuleManager
* launch ship with kOS core on it
* >> switch to 0.
* >> run tmp.ks.

Edit: the file compiles fine, and when running the compiled version, I get the error once without having to restart the game every time.

**Comment by pellinor0** [comment]
I could boil it down further: there seems to be a problem with the symbol "d". The code below triggers the errors. Once I declare the first d as a "local function", things work again. 
```
@lazyglobal off.
function timeToDist {
    parameter dist.
    function d {parameter t. return dist-(PositionAt(Ship,t)-PositionAt(Target,t)):Mag.}
}

function findClosestApproach {
    local d is 0.
    local t is 0.
    set d to Abs( (PositionAt(Ship,t)-PositionAt(Target,t)):Mag ).
    return 0.
}
```

**Comment by hvacengi** [comment]
Apologies, this issue got lost in the holiday shuffle.  I think that this is probably related to #2204 (which was posted after yours).  As you correctly found, the `local` keyword is the key.  The new version apparently introduced a bug where nested functions are not properly being compiled with local scope and instead are being defined in global scope.

This also points out however that we appear to have an issue in the error reporting code, and that we need to make the error more clear.  Unfortunately your two uploaded files appear to have expired, at least they are returning only the text `{"success":false,"error":404,"message":"Not Found"}` for me.

**Comment by Dunbaratu** [comment]
@pellinor0 - I was trying to re-create this problem to test it, and I cannot find your code examples.  The links you provided in the original post lead to 404 Not Found errors.

**Comment by pellinor0** [comment]
Sorry for the confusion. The second post was meant as an update of the first one, so the links are not needed anymore. 

You should only need the few lines of code from the second post. I got the error just by running a file with that content. It happened once when running the compiled file, and in a game-breaking loop when running it uncompiled.

**Comment by Dunbaratu** [comment]
I suspect the PR I'm making to fix issue #2204 will also fix this problem.  I'll test afterward to see.

**Comment by Dunbaratu** [comment]
Okay I was able to reproduce it, and verify that my fix to issue 2204 also fixes this.

But I also want to make the error message better in case it happens again - so it doesn't get stuck in a loop printing the error again and again like it did here.

The irony is that this means I have to fix the error message first as a separate PR *without* fixing the root problem, because once I fix this problem, the error message won't be triggered.
